compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1127,9 +1127,7 @@ fn link_natively(
         let stripcmd = if !sess.host.is_like_solaris { "rust-objcopy" } else { "/usr/bin/strip" };
         match strip {
             // Always preserve the symbol table (-x).
-            Strip::Debuginfo => {
-                strip_symbols_with_external_utility(sess, stripcmd, out_filename, &["-x"])
-            }
+            Strip::Debuginfo => strip_with_external_utility(sess, stripcmd, out_filename, &["-x"]),
```

Comment:
Suggestion [DISCARD-ALL 2/2]: we should also be able to spell out `--discard-all`, AFAICT.

Reply:
I would prefer to not change the -x, to keep the changes minimal.

Reply:
That's fine (if this is intended for a fix-forward to potentially backport to address the regression)

Reply:
(Filed #135038 as an easy follow-up)